pkg/asset/installconfig/aws/validation.go

```diff
@@ -471,53 +485,48 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets Subnets, idxMap m
 
 func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) field.ErrorList {
 	allErrs := field.ErrorList{}
-	ec2Endpoint := ""
+	endpointOverrides := make(map[string]string)
 	for id, service := range services {
-		err := validateEndpointAccessibility(service.URL)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Index(id).Child("url"), service.URL, err.Error()))
-			continue
-		}
-		if service.Name == ec2.ServiceName {
-			ec2Endpoint = service.URL
-		}
-	}
-
-	if partition, partitionFound := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); partitionFound {
-		if _, ok := partition.Regions()[region]; !ok && ec2Endpoint == "" {
-			err := validateRegion(region)
+		if !sets.New(requiredServices...).Has(service.Name) {
+			err := validateEndpointAccessibility(service.URL)
 			if err != nil {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("region"), region, err.Error()))
+				allErrs = append(allErrs, field.Invalid(fldPath.Index(id).Child("url"), service.URL, err.Error()))
 			}
 		}
-		return allErrs
+		endpointOverrides[service.Name] = service.URL
 	}
 
-	resolver := newAWSResolver(region, services)
 	var errs []error
+	// validate the service endpoint for each of the required services. This will be the default endpoint for
+	// services where the user did not override the service endpoint through the install-config.
 	for _, service := range requiredServices {
-		_, err := resolver.EndpointFor(service, region, endpoints.StrictMatchingOption)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("failed to find endpoint for service %q: %w", service, err))
+		if urlOverride, ok := endpointOverrides[service]; ok {
+			if err := validateEndpointAccessibility(urlOverride); err != nil {
+				logrus.Debugf("failed to access %s endpoint at %s", service, urlOverride)
+				errs = append(errs, field.Invalid(fldPath, region, fmt.Sprintf("failed to access %s endpoint override %s", service, urlOverride)))
+			}
+		} else {
+			defaultEndpoint, err := getDefaultServiceEndpoint(region, service, endpointOptions{
+				DisableHTTPS:         false,
+				UseDualStackEndpoint: awsv2.DualStackEndpointStateDisabled,
+			})
+			if err != nil {
+				errs = append(errs, field.Invalid(fldPath, region, fmt.Sprintf("could not determine %s endpoint for region %s", service, region)))
+			} else {
+				if err := validateEndpointAccessibility(defaultEndpoint.URL); err != nil {
+					logrus.Debugf("failed to access %s endpoint at %s", service, defaultEndpoint.URL)
+					errs = append(errs, field.Invalid(fldPath, region, fmt.Sprintf("failed to access default %s endpoint", service)))
+				}
+			}
```

Comment:
Adding to the other thread https://github.com/openshift/installer/pull/9383#discussion_r1934803589 I don't think it's necessary to validate the default endpoints. The code looks good, but it would be simpler to omit the default checks.